File: OSA.Infructructure.Services/EmployeeService.cs
using OAS.Core.Entity;
using OSA.Core.Interface;
using OSA.Infructructure.Services.Base;
using System;
using System.Collections.Generic;
using System.Text;

namespace OSA.Infructructure.Services
{
    public class EmployeeService : BaseService<Employee> , IEmployeeRepository
    {
    }
}

File: Angular/src/app/Shared/Api/employee.service.ts
import { Injectable } from '@angular/core';
import { BaseService } from './Base/base-service.service';
import { Employee } from '../Models/Employee';
import { IEmployeeRepository } from './iemployee-repository';
import { HttpClient } from '@angular/common/http';

@Injectable({
  providedIn: 'root'
})
export class EmployeeService extends BaseService<Employee> implements IEmployeeRepository {

  constructor(private httpp : HttpClient) {
    super(httpp,"https://localhost:44370/api/Employees")//, new Employee());
  }
 
}
